docs/proposals/PROPOSAL_H.md

```diff
@@ -0,0 +1,359 @@
+
+```json
+{
+  "schemaVersion": "0.1.0",
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.oci.image-compatibility.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710,
+      "annotations": {
+        "name": "profileH"
+      }
+    }
+  ],
+  "subject": {
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
+    "size": 7682
+  },
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-03-27T08:08:08Z"
+  }
+}
+```
+
+The new artifact type for image compatibility MUST be referenced in OCI documentation.
+
+The schema version is to use string semver instead of int version, to help tool in content deserialization.
+
+The media type is defined as vnd.oci.image-compatibility.spec, in order to be defined agnostic to cluster management software.
+
+### Image Compatibility Spec
+
+The spec structure is quite simple, a flat json file.
+
+Schema:
+
+- **`compatibilities`** _list of compatibility sets_
+
+  This REQUIRED property describes all image compatibility sets, contains at least one compatibility set.
+
+  - **`compatibility set`** _list of compatibility requirements_
+
+    This REQUIRED property describes compatibility requirements in one compatibility set.
+
+    If one of the compatibility sets passes the compatibility check against the target node, that means the node is compatible with the image.
+
+    There are three category key-value pair inside the compatibility set.
+
+    - **`compatibility feature field`**
+
+      This REQUIRED property describes a feature compatibility requirement on the target host, the field accepts OCI and community defined labels.
+
+      The value of compatibility field in the compatibility artifact SHOULD be described in contiguous or discontinuous range if the value discovered on nodes can be different.
+
+      It's the tool's responsibility to interpret the value of OCI or community defined feature label, and matching criteria with the value discovered from the node via NFD tool or other tool. NFD means the Kubernetes node feature discovering SIG.
+
+      Inevitably, some compatibility fields MAY be duplicated among compatibility sets.
+
+      The value type of this field is STRING.
+
+    - **`compatibility score field`**
+
+      This OPTIONAL property describes the score of the compatibility set.
+
+      There is only one compatibility score field allowed in one compatibility set.
+
+      The highest score of the compatibility set means this compatibility set is the most optimal one.
```

Comment:
It's not clear to me still how you'd compare between compatibility artifacts. This (still) feels like something dangerous to provide or mention. If a community wants to add a score to their label schema we can't stop them, but I don't see the value to make it an optional field (and suggest it's a good idea). I would argue minimally we remove this for now and only consider it in the future when it's absolutely needed, or call it out of scope for now and consider it part of the next discussion about labels.

Reply:
The other issue with the score is that it's not really about compatibility, it's going into optimization "works best" and not just "it works."

Reply:
> It's not clear to me still how you'd compare between compatibility artifacts. This (still) feels like something dangerous to provide or mention.

I don't think it's about comparing compatibility artifacts, but about finding the best matching node within a single artifact.

Reply:
I think optimization is out of scope for compatibility, and a specific label is out of scope for the structure of the artifact discussion.

Reply:
> It's not clear to me still how you'd compare between compatibility artifacts. This (still) feels like something dangerous to provide or mention. If a community wants to add a score to their label schema we can't stop them, but I don't see the value to make it an optional field (and suggest it's a good idea). I would argue minimally we remove this for now and only consider it in the future when it's absolutely needed, or call it out of scope for now and consider it part of the next discussion about labels.

agree, labels deicussion could be in next phase